### PR TITLE
Merge ui test cases referring same action (closes #843)

### LINKED ISSUE #843: Merge current/non-current branch UI test cases
From https://github.com/VirtusLab/git-machete-intellij-plugin/pull/823#discussion_r872976921.

That should save us some time, as setting up a new IDE instance & sample git repo for each test **case** (i.e. each test **method**) constitutes a major overhead.

### PR DESCRIPTION
**Pull request by mkondratek**

### CODE REVIEW

**Comment by mkondratek** [comment]
Heh, it passed once but now the same code fails due to `java.net.SocketTimeoutException: timeout` 🙄

**Comment by PawelLipski** [comment]
Huh looks that the help of PR #922 would be useful ;D

File: src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala [hunk 1]
```diff
@@ -187,26 +188,27 @@ class UITestSuite extends BaseGitRepositoryBackedIntegrationTestSuite(SETUP_WITH
     project.assertNoUncommittedChanges()
   }
 
-  @Test def resetCurrentBranchToRemote(): Unit = {
+  @Test def resetBranchToRemote(): Unit = {
```
[the remark this answers is not in the document]
Comment (mkondratek) [reply src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala:191]: I did not check every build but
- at least a few of them failed at reset to remote
- I do not recall any other test failing

File: src/uiTest/resources/project.rhino.js [hunk 2]
```diff
@@ -282,18 +282,28 @@ function Project(underlyingProject) {
...
   const clickMouseInTheMiddle = function () {
-    const ideFrame = getComponentByClass('com.intellij.openapi.wm.impl.IdeFrameImpl');
+    const ideFrame = getIdeFrame();
     robot.click(ideFrame);
   };
 
+  const getIdeFrame = function() {
+    return getComponentByClass(
+        /* className */ 'com.intellij.openapi.wm.impl.IdeFrameImpl',
+        /* text */ 'machete-sandbox-worktree',
```
Comment (PawelLipski) [inline src/uiTest/resources/project.rhino.js:300]: Would it make sense to just squeeze the two params into a `predicate` in `getComponentByClass`?
Like 
```
/* predicate */ component => 'machete-sandbox-worktree'.equals(component.getTitle())
```

Comment (mkondratek) [reply src/uiTest/resources/project.rhino.js:300]: good idea

File: src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala [hunk 3]
```diff
@@ -36,7 +36,6 @@ class UITestSuite extends BaseGitRepositoryBackedIntegrationTestSuite(SETUP_WITH
     waitAndCloseProject()
   }
 
-  @Ignore // FIXME (#830): ui test failing - socket timeout
```
Comment (PawelLipski) [inline src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala:39]: Wow actually... so your changes from this PR are enough to fix all the flakiness? 😯

Comment (PawelLipski) [reply src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala:39]: I.e. just selecting the right `IdeFrameImpl`

File: src/uiTest/resources/project.rhino.js [hunk 4]
```diff
@@ -22,15 +22,32 @@ importClass(org.assertj.swing.fixture.JTableFixture);
 importClass(org.assertj.swing.data.TableCell);
 
 
+let myClick = function (component, mouseButton) {
```
Comment (mkondratek) [inline src/uiTest/resources/project.rhino.js:25]: btw, for UITestSuite there is no need for "pretty click". What more it's even better to speed up the process and click the buttons ASAP. That will save us some time too

File: src/uiTest/resources/project.rhino.js [hunk 5]
```diff
@@ -297,16 +301,17 @@ function Project(underlyingProject) {
   const getIdeFrame = function() {
     return getComponentByClassAndText(
         /* className */ 'com.intellij.openapi.wm.impl.IdeFrameImpl',
-        /* text */ 'machete-sandbox-worktree',
-        /* textCmp */function (text, component) {
-          return text.equals(component.getTitle());
-        }
+        // Using "includes" as in UITestSuite the project title is "machete-sandbox-worktree"
+        // but in UIScenarioSuite it's "machete-sandbox"
+        /* predicate */ component => 'machete-sandbox-worktree'.includes(component.getTitle())
```
Comment (PawelLipski) [inline src/uiTest/resources/project.rhino.js:306]: Hmmm and if the title is empty (as used to be for the other `IdeFrameImpl`s)?
Shouldn't it be:
```suggestion
        /* predicate */ component => component.getTitle().contains('machete-sandbox')
```

File: src/uiTest/resources/project.rhino.js [hunk 6]
```diff
@@ -297,16 +301,17 @@ function Project(underlyingProject) {
...
     );
   }
 
   const findAndClickButton = function (name) {
-    const button = getComponentByClassAndText('javax.swing.JButton', name);
-    prettyClick(button, MouseButton.LEFT_BUTTON);
+    const button = getComponentByClassAndText('javax.swing.JButton',
```
Comment (PawelLipski) [inline src/uiTest/resources/project.rhino.js:311]:
```suggestion
    const button = getComponentByClassAndPredicate('javax.swing.JButton',
```

File: src/uiTest/resources/project.rhino.js [hunk 7]
```diff
@@ -98,10 +115,7 @@ function Project(underlyingProject) {
 
 
   // Actions
-
-  const ACTION_PLACE_TOOLBAR = 'GitMacheteToolbar';
   const ACTION_PLACE_CONTEXT_MENU = 'GitMacheteContextMenu';
```
Comment (PawelLipski) [inline src/uiTest/resources/project.rhino.js:118]: Still used?

File: src/uiTest/resources/project.rhino.js [hunk 8]
```diff
@@ -98,10 +115,7 @@ function Project(underlyingProject) {
...
-  const RESET_INFO_SHOWN = 'git-machete.reset.info.shown';
 
   const getActionByName = function (actionName) {
```
Comment (PawelLipski) [inline src/uiTest/resources/project.rhino.js:120]: Ditto, still used?

File: src/uiTest/resources/project.rhino.js [hunk 9]
```diff
@@ -22,15 +22,32 @@ importClass(org.assertj.swing.fixture.JTableFixture);
...
+    robot.moveMouse(component);
+    // Wait for a while before clicking to allow the scenario spectator to see the button being clicked
+    for (let i = 0; i < 5; i++) {
+      sleep();
+    }
+    robot.click(component, mouseButton);
+  };
+
+  this.usePrettyClick = function () {
+    myClick = prettyClick;
```
Comment (PawelLipski) [inline src/uiTest/resources/project.rhino.js:45]: Almost like virtual methods :js: :poorly-designed-spurdo:

File: src/uiTest/resources/project.rhino.js (hunk 7 above)
Comment (mkondratek) [reply src/uiTest/resources/project.rhino.js:118]: yes

File: src/uiTest/resources/project.rhino.js (hunk 9 above)
Comment (mkondratek) [reply src/uiTest/resources/project.rhino.js:45]: yes 🤓

File: src/uiTest/resources/project.rhino.js [hunk 10]
```diff
@@ -22,15 +22,32 @@ importClass(org.assertj.swing.fixture.JTableFixture);
...
 function Project(underlyingProject) {
 
   const pluginId = PluginId.getId('com.virtuslab.git-machete');
   const pluginClassLoader = PluginManagerCore.getPlugin(pluginId).getPluginClassLoader();
 
+  const prettyClick = function (component, mouseButton) {
+    robot.moveMouse(component);
+    // Wait for a while before clicking to allow the scenario spectator to see the button being clicked
+    for (let i = 0; i < 5; i++) {
+      sleep();
```
[the remark this answers is not in the document]
Comment (mkondratek) [reply src/uiTest/resources/project.rhino.js:39]: ok

File: src/uiTest/resources/project.rhino.js [hunk 11]
```diff
@@ -42,7 +59,7 @@ function Project(underlyingProject) {
   // Tab & model management
   this.openGitMacheteTab = function () {
     openTab(ToolWindowId.VCS, 'Git Machete');
-  }
+  };
```
[the remark this answers is not in the document]
Comment (mkondratek) [reply src/uiTest/resources/project.rhino.js:63]: https://dev.to/adriennemiller/semicolons-in-javascript-to-use-or-not-to-use-2nli

probably not xd

[the remark this answers is not in the document]
Comment (PawelLipski) [reply src/uiTest/resources/project.rhino.js:63]: Okay, so similar to Scala semicolon inference... although apparently **less** keen on adding semicolons than Scala ;p

**Review by PawelLipski — APPROVED** [review]
_Alles gut_!

**Pull request merged**